src/main/scala/sigmastate/lang/SigmaBuilder.scala

```diff
@@ -472,8 +472,10 @@ class StdSigmaBuilder extends SigmaBuilder {
   override def mkZKProofBlock(body: Value[SSigmaProp.type]): BoolValue =
     ZKProofBlock(body)
 
-  override def mkVal(name: String, givenType: SType, body: Value[SType]): Val =
-    ValNode(name, givenType, body)
+  override def mkVal(name: String,
+                     givenType: SType,
+                     body: Value[SType], srcCtx: SourceContext): Val =
+    ValNode(name, givenType, body, Nullable(srcCtx))
```

Comment:
As discussed above, don't need to change builder API

Reply:
Done.